helix-term/src/commands.rs

```diff
@@ -1811,6 +1812,20 @@ fn select_all(cx: &mut Context) {
     doc.set_selection(view.id, Selection::single(0, end))
 }
 
+fn select_first_and_last_chars(cx: &mut Context) {
+    let (view, doc) = current!(cx.editor);
+
+    let selection = doc.selection(view.id).clone().transform_iter(|range| {
+        vec![
+            Range::new(range.anchor, range.anchor + 1),
+            Range::new(range.head - 1, range.head),
```

Comment:
The subtraction here can cause a crash: open a scratch buffer and delete the trailing newline so the buffer is empty, then `A-S` subtracts with overflow.

Instead let's use points and the prev_grapheme_boundary helper:

```suggestion
            Range::new(range.from()),
            Range::new(graphemes::prev_grapheme_boundary(text, range.to())),
```